### PR TITLE
Add autocomplete screen reader support

### PR DESCRIPTION
**Pull request by nmanu1**
Add a `ScreenReader` component that is used to provide screen reader instructions and announcements. Use this component in `InputDropdown` to give autocomplete instructions and screen reader announcements for number of autocomplete options found.

J=SLAP-1402
TEST=manual

Check that screen reader instructions and announcements are correct in the sample app. An announcement is made for each input change and on clicking the input bar if there is input present.

### CODE REVIEW

File: sample-app/src/components/InputDropdown.tsx [hunk 1]
```diff
@@ -1,9 +1,14 @@
...
 import Dropdown, { Option } from './Dropdown';
+import ScreenReader from "./ScreenReader";
+import { processTranslation } from './utils/processTranslation';
 
 interface Props {
   inputValue?: string,
   placeholder?: string,
+  screenReaderInstructions?: string,
+  screenReaderInstructionsId?: string,
+  shouldAutocompleteCount: boolean,
```
[the remark this answers is not in the document]
Comment (oshi97) [reply sample-app/src/components/InputDropdown.tsx:11]: sgtm!

File: sample-app/src/components/InputDropdown.tsx [hunk 2]
```diff
@@ -61,14 +69,19 @@ export default function InputDropdown({
     focusedOptionIndex: undefined,
     shouldDisplayDropdown: false,
   });
-  const focusOptionId = focusedOptionIndex === undefined 
+  const focusOptionId = focusedOptionIndex === undefined
     ? undefined
     : `${optionIdPrefix}-${focusedOptionIndex}`;
 
   const [latestUserInput, setLatestUserInput] = useState(inputValue);
+  const [countKey, setCountKey] = useState(0);
```
[the remark this answers is not in the document]
Comment (oshi97) [reply sample-app/src/components/InputDropdown.tsx:77]: Oh I didn't realize we were using 0 as "don't render", lgtm!

[the remark this answers is not in the document]
Comment (oshi97) [reply sample-app/src/components/InputDropdown.tsx:77]: I see I didn't realize countKey === 0 was being used as "dont render", don't need to useReducer then!

File: sample-app/src/components/InputDropdown.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (oshi97) [reply sample-app/src/components/InputDropdown.tsx:11]: I like making screenReaderInstructions and screenReaderInstructionsId required, and also removing shouldAutocompleteCount and shouldCount

[the remark this answers is not in the document]
Comment (nmanu1) [reply sample-app/src/components/InputDropdown.tsx]: updated!

**Review by oshi97 — APPROVED** [review]
looks great!

**Review by cea2aj — APPROVED** [review]
🔥

**Pull request merged**